test/bridge_test.rb

```diff
@@ -6,87 +6,167 @@
-          "tun0" => {
-            "BOOTPROTO" => "static",
-            "STARTMODE" => "onboot",
-            "TUNNEL"    => "tun"
-          }
-        },
-        "tap"  => {
-          "tap0" => {
-            "BOOTPROTO" => "static",
-            "STARTMODE" => "onboot",
-            "TUNNEL"    => "tap"
-          }
-        },
-        "br"   => {
-          "br0" => { "BOOTPROTO" => "dhcp" }
+describe Yast::LanItems do
+  let(:netconfig_items) do
+    {
+      "eth"  => {
+        "eth1" => { "BOOTPROTO" => "none" },
+        "eth2" => { "BOOTPROTO" => "none" },
+        "eth4" => {
+          "BOOTPROTO" => "static",
+          "IPADDR"    => "0.0.0.0",
+          "PREFIX"    => "32"
         },
-        "bond" => {
-          "bond0" => {
-            "BOOTPROTO"      => "static",
-            "BONDING_MASTER" => "yes",
-            "BONDING_SLAVE0" => "eth1",
-            "BONDING_SLAVE1" => "eth2"
-          }
+        "eth5" => { "BOOTPROTO" => "static", "STARTMODE" => "nfsroot" },
+        "eth6" => { "BOOTPROTO" => "static", "STARTMODE" => "ifplugd" }
+      },
+      "tun"  => {
+        "tun0" => {
+          "BOOTPROTO" => "static",
+          "STARTMODE" => "onboot",
+          "TUNNEL"    => "tun"
+        }
+      },
+      "tap"  => {
+        "tap0" => {
+          "BOOTPROTO" => "static",
+          "STARTMODE" => "onboot",
+          "TUNNEL"    => "tap"
+        }
+      },
+      "br"   => {
+        "br0" => { "BOOTPROTO" => "dhcp" }
+      },
+      "bond" => {
+        "bond0" => {
+          "BOOTPROTO"      => "static",
+          "BONDING_MASTER" => "yes",
+          "BONDING_SLAVE0" => "eth1",
+          "BONDING_SLAVE1" => "eth2"
         }
       }
+    }
+  end
+
+  let(:hwinfo_items) do
+    [
+      { "dev_name" => "eth11" },
+      { "dev_name" => "eth12" }
+    ]
+  end
+
+  let(:expected_bridgeable) do
+    [
+      "bond0",
+      "eth4",
+      "eth11",
+      "eth12",
+      "tap0"
+    ]
+  end
+
+  before(:each) do
+    allow(Yast::NetworkInterfaces).to receive(:FilterDevices).with("netcard") { netconfig_items }
+
+    allow(Yast::LanItems).to receive(:ReadHardware) { hwinfo_items }
+
+    Yast::LanItems.Read
+  end
+
+  describe "#GetBridgeableInterfaces" do
+    before(:each) do
+      # FindAndSelect initializes internal state of LanItems it
+      # is used internally by some helpers
+      Yast::LanItems.FindAndSelect("br0")
     end
 
-    let(:hwinfo_items) do
-      [
-        { "dev_name" => "eth11" },
-        { "dev_name" => "eth12" }
-      ]
+    it "returns list of slave candidates" do
+      expect(
+        Yast::LanItems
+          .GetBridgeableInterfaces(Yast::LanItems.GetCurrentName)
+          .map { |i| Yast::LanItems.GetDeviceName(i) }
+      ).to match_array expected_bridgeable
     end
+  end
 
-    let(:expected_bridgeable) do
-      [
-        "bond0",
-        "eth4",
-        "eth11",
-        "eth12",
-        "tap0"
-      ]
+  describe "#old_bridge_port_config?" do
```

Comment:
please add a test for the "eth1" case , which has `  { "BOOTPROTO" => "none" } ` and so should return false